### ISSUE TITLE
[Documentation] Support for .cgi files that are ruby files?

### ISSUE THREAD

**Issue by rubyFeedback** [puma/puma#2878]
Hey puma devs, this is a documentation request.

Background: I started with PHP in 2003 or something like that. Then I moved to ruby.

I still used .cgi files - and still use them today. So I need a webserver that works with
.cgi files that serve ruby content (the .cgi file is a ruby file really).

I solve this on linux fine via lighttpd. On Windows I could use XAMP, but apache
goes on my nerves way too much. I hate the config format it uses. (I autogenerate
my config for lighttpd, so I can adapt quickly from the commandline.)

My .cgi files are actually not even "pure" ruby files anymore; they use a central
file that is then served via e. g. the .cgi file or also a .sinatra file. So I am already
using puma as it is and it works well. Never had any big issues with it + sinatra.

But I also have a few hundred legacy .cgi files, and for these I'd
need a way to quickly serve them (transitioning to my hybrid sinatra setup takes
a few minutes per .cgi file, I also have to check that things work after I ported
this), AND - here is the real reason why I am filing this - I need simple windows
support out of the box. Windows is annoying me a LOT. I have no patience with
it. :(

Now I already tested my sinatra files all work fine on windows too, so this is
Plan B - I could transition everything into my hybrid setup. This is what I
plan to do in the long run, and add rails support as well via a huge monolith
that manages all my knowledge base. But for the time being I need simple
support for .cgi files to get in dynamic content. (Sinatra also annoys me but
I digress.)

Anyway - so much for the intro and background. **Now to my issue request,
TL;DR**:

- Does puma work with raw, simple .cgi files for ruby? And if so, how is this
done? Would it be possible to add a simple example showcasing this 
including instructions in the comment of the .cgi file? This can really be
a simple "hello world".

For instance:

    require 'cgi'

    alias e puts

    cgi = CGI.new 'html4' # Create a cgi object, with HTML 4 generation methods.
    e cgi.header

    e '<html><body>This is a test.<br><br>'
    if cgi.query_string and !cgi.query_string.empty?
      e 'The query string is: <b>'+cgi.query_string+'</b><br><br>'
    end
    e 'The header was:'
    e cgi.header
    e '</body></html>'

This works on my lighttpd setup but I have no idea how I can let puma 
handle this. Can I use puma like apache or lighttpd? If so how and where
to look for? Could the main README also mention "cgi" so that it is easier
to search for? Or alternatively, if my above use case does not work, then
to mention this?

I hope I could explain my use case. Right now I am a bit confused, and
one problem with ruby software in general is lack of documentation
(compare to lighttpd or apache documentation available on the www,
we ruby guys kind of lose out in the information segment.)

**Comment by MSP-Greg** [comment]
I don't think this is Puma related.  If you can create a 'rack app' that does the processing, Puma or other Ruby app servers should be able to process it?  Or, see [Sinatra info](https://msp-greg.github.io/puma/index.html#sinatra)?

**Comment by nateberkopec** [comment]
Puma is a Rack application server and does not support CGI.

[Webrick does support CGI](https://www.mikeperham.com/2018/10/12/testing-rubys-cgi/), however. Give that a shot.
